src/Datadog.Trace.ClrProfiler.Managed/Integrations/Caching/InterceptedMethodAccess.cs

```diff
@@ -50,5 +50,34 @@ internal class InterceptedMethodAccess<TDelegate>
                             genericTypes: generics);
                     });
         }
+
+        /// <summary>
+        /// Attempts to retrieve a method from cache, otherwise creates the delegate reference, adds it to the cache, and then returns it.
+        /// </summary>
+        /// <param name="owningType">Type which owns the method.</param>
+        /// <param name="methodName">Name of the method being instrumented.</param>
+        /// <param name="returnType">The return type of the instrumented method.</param>
+        /// <param name="generics">The ordered types of the method's generics.</param>
+        /// <param name="parameters">The ordered types of the method's parameters.</param>
+        /// <returns>Delegate representing instrumented method.</returns>
+        internal TDelegate GetInterceptedMethod(
```

Comment:
How come we need both of these methods, one with and one without the assembly? Can we patch the AspNetCoreMvc2Integration Rethrow method to consume this new GetInterceptedMethod?

Reply:
Absolutely. I'll hit that in another PR.